### PR TITLE
blog page and blog landing page

### PR DESCRIPTION
**Pull request by sarahframe**
I've added a field called 'category' to be used in the blogpost tile with max char limit of 20 @liamjohnston

### CODE REVIEW

File: cdhweb/blog/views.py [hunk 1]
```diff
@@ -89,30 +89,10 @@ class BlogDetailView(BlogPostMixinView, DetailView, LastModifiedMixin):
...
-            prev = prev[0]
-        else:
-            prev = None
-
-        context.update({"opengraph_type": "article", "next": next, "previous": prev})
-        return context
+    def get(self, request, *args, **kwargs):
+        self.object = self.get_object()
+
+        return self.object.serve(request, *args, **kwargs)
```
Comment (haydngreatnews) [inline cdhweb/blog/views.py:95]: :+1:

File: cdhweb/blog/models.py [hunk 2]
```diff
@@ -54,40 +58,106 @@ class BlogPostTag(TaggedItemBase):
...
+        blank=True,
+        null=True,
+        features=["bold", "italic"],
+        verbose_name="Blog description",
+        help_text="Short introduction to the page, aim for max two clear sentences (max. 200 chars).",
+    )
+
+    article_date = models.DateTimeField(
+        verbose_name="Published date",
+        default=localdate,
```
Comment (haydngreatnews) [inline cdhweb/blog/models.py]: I'm a bit wary of `localdate` here -- This returns a python `date` object, and they cant have timezones, so I'm not entirely sure what timezones might do to this when it's saved to the DB as a zone-aware timestamp.

Out of curiosity, I tried it out, and got:
```python
In [5]: from django.utils.timezone import localdate

In [6]: t.start_time = localdate()

In [7]: t.save()
/home/haydn/.pyenv/versions/legend-story-app/lib/python3.11/site-packages/django/db/models/fields/__init__.py:1535: RuntimeWarning: DateTimeField Tournament.start_time received a naive datetime (2024-06-12 00:00:00) while time zone support is active.

In [8]: t.refresh_from_db()

In [9]: t.start_time
Out[9]: datetime.datetime(2024, 6, 11, 12, 0, tzinfo=datetime.timezone.utc)

In [10]: localdate()
Out[10]: datetime.date(2024, 6, 12)

```
So it does look like (in our timezone, admittedly) like the resulting DB column can end up with a wrong-ish value.

I think we should either make this a `DateField`, which will just store the date, and be deliberately ignorant of timezones (which I think is fine for this usecase), or we should be storing a datetime from `timezone.now()` (also, you can probably [use `auto_now_add=True` to take care of the default](https://docs.djangoproject.com/en/5.0/ref/models/fields/#django.db.models.DateField.auto_now_add), for either case)

File: cdhweb/blog/models.py [hunk 3]
```diff
@@ -54,40 +58,106 @@ class BlogPostTag(TaggedItemBase):
...
+    )
     tags = ClusterTaggableManager(through=BlogPostTag, blank=True)
     featured = models.BooleanField(
         default=False, help_text="Show the post in the carousel on the homepage."
     )
     people = models.ManyToManyField(Person, through="blog.Author", related_name="posts")
 
     # can only be created underneath special link page
-    parent_page_types = ["blog.BlogLinkPage"]
+    parent_page_types = ["blog.BlogLinkPageArchived", "blog.BlogLandingPage"]
```
Comment (haydngreatnews) [inline cdhweb/blog/models.py:125]: You could take the archived one out of here, if you don't want them to add any blogs to the old one (up to you whether that's the right thing or not though)

File: cdhweb/blog/models.py [hunk 4]
```diff
@@ -104,33 +174,35 @@ class BlogPost(BasePage, ClusterableModel, PagePreviewDescriptionMixin):
...
+    #     return Truncator(self.title).chars(65)
 
-    @property
-    def short_title(self):
-        """Shorter title with ellipsis."""
-        return Truncator(self.title).chars(65)
+    # @property
+    # def short_description(self):
+    #     """Shorter description with ellipsis."""
+    #     return Truncator(self.get_description()).chars(250)
```
Comment (haydngreatnews) [inline cdhweb/blog/models.py]: Are we ready to remove these?

File: cdhweb/blog/models.py [hunk 5]
```diff
@@ -171,3 +243,15 @@ class BlogLinkPage(LinkPage):
     # NOTE the only allowed child page type is a BlogPost; this is so that
     # Events made in the admin automatically are created here.
     subpage_types = [BlogPost]
+
+
+class BlogLandingPage(Page, StandardHeroMixinNoImage):
```
Comment (haydngreatnews) [inline cdhweb/blog/models.py]: We should probably switch the order of these parent classes -- methods (and other things, i imagine) are resolved left-to-right, so anything `Page` implements will override anything `StandardHeroMixinNoImage` implements.

For example, if `StandardHeroMixin` ever overrode `get_context` it wouldn't come through here

File: cdhweb/blog/tests/conftest.py [hunk 6]
```diff
@@ -4,15 +4,15 @@
 import pytest
 from django.utils import timezone
 
-from cdhweb.blog.models import Author, BlogLinkPage, BlogPost
+from cdhweb.blog.models import Author, BlogLinkPageArchivedArchived, BlogPost
```
Comment (haydngreatnews) [inline cdhweb/blog/tests/conftest.py]: Extra `Archived` here?

File: cdhweb/blog/models.py (hunk 3 above)
Comment (sarahframe) [reply cdhweb/blog/models.py:125]: I was planning on leaving these in until I move the pages to the new landing page type

File: cdhweb/blog/models.py (hunk 2 above)
Comment (sarahframe) [reply cdhweb/blog/models.py]: I've removed this field now as it's not necessary. They were originally using 'first_published_by' so going to continue to use that which will help with the data migration too

**Review by haydngreatnews — APPROVED** [review]

**Pull request merged**